renderdoc/serialise/serialiser.cpp

```diff
@@ -271,6 +280,8 @@ uint32_t Serialiser<SerialiserMode::Writing>::BeginChunk(uint32_t chunkID, uint3
       RDCASSERT(chunkID <= ChunkIndexMask);
 
       c |= m_ChunkFlags;
+      if(byteLength > uint64_t(std::numeric_limits<uint32_t>::max()))
```

Comment:
I find plain `0xFFFFFFFF` here more readable than `numeric_limits` etc. Unlike FLT_MAX or something it's immediately obvious what's being checked especially with the flag name.

Reply:
Done. I've been burnt by lots of ffff in constants like that (people writing 0xfffffff [missing one 'f']). C++14's digit separator does help ensure the proper values are typed (0xffff'ffff and 0x1'0000'0000).

Do you support C++14 at all yet? Or still have some platform that are restricted to C++11/0x?

Reply:
That's a fair concern, though hopefully something that can be caught in review. Where possible I might use `~0U` for something like this but my intuition is not to do that when mixing and matching between 32-bit and 64-bit since then it's quite confusing too.

The UI uses C++14 for a couple of tiny bits of code in the python bindings, but I only use C++11 in the core code. I haven't tested it but my assumption is that Android probably can't handle it because Android is completely terrible at everything.